feat(rslint_parser): Parse 2nd param for JsImportCallExpression

Boshen opened the pull request:
## Summary
This PR changes dynamic import to accept a second param only.

Parsing just two parameters should be simpler than calling `parse_arguments` and then check the negative cases.

## Test Plan
Lots of test262 passes regarding dynamic import, the remaining failed tests has to do with negative tests around `new import('foo')` and `await import('foo')`

Boshen commented:
From local run:

| Test result | `main` count | This PR count | Difference |
| :---------: | :----------: | :-----------: | :--------: |
| Total | 44951 | 44951 | 0 |
| Passed | 42892 | 42954 | ✅ ⏫ **+62** |
| Failed | 2058 | 1996 | ✅ ⏬ **-62** |
| Panics | 1 | 1 | 0 |
| Coverage | 95.42% | 95.56% | **+0.14%** |

<details><summary><b>:tada: Fixed (62):</b></summary>

```
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-assert-enumeration-abrupt.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-assert-enumeration.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-assert-non-object.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-assert-undefined.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-assert-value-abrupt.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-assert-value-non-string.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-await-expr.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-await-ident.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-evaluation-abrupt-return.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-evaluation-abrupt-throw.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-evaluation-sequence.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-get-assert-error.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-non-object.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-trailing-comma-fulfill.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-trailing-comma-reject.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-yield-expr.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/2nd-param-yield-ident-valid.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-arrow-assignment-expression-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-arrow-assignment-expression-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-arrow-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-arrow-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-arrow-function-await-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-arrow-function-await-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-arrow-function-return-await-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-arrow-function-return-await-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-function-await-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-function-await-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-function-return-await-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-function-return-await-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-function-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-function-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-gen-await-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-async-gen-await-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-block-labeled-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-block-labeled-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-block-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-block-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-do-while-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-do-while-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-else-braceless-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-else-braceless-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-else-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-else-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-function-return-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-function-return-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-function-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-function-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-if-braceless-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-if-braceless-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-if-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-if-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-while-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-while-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-with-expression-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-with-expression-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-with-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/nested-with-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/top-level-trailing-comma-first.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/syntax/valid/top-level-trailing-comma-second.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/trailing-comma-fulfill.js
xtask/src/coverage/test262/test/language/expressions/dynamic-import/trailing-comma-reject.js
xtask/src/coverage/test262/test/language/import/json-idempotency.js
```
</details>

MichaReiser commented on crates/rslint_parser/test_data/inline/err/import_call_no_arg.rast:
```diff
@@ -1,93 +0,0 @@
-      1: SEMICOLON@23..24 ";" [] []
-  3: EOF@24..25 "" [Whitespace("\n")] []
---
-error[SyntaxError]: expected an expression, or an assignment but instead found ')'
```
Can we keep at least one test, testing if at least one argument is passed?

Also, can we add tests, veryfing that passing more than 2 arguments isn't valid either.

MichaReiser commented on crates/rslint_parser/src/syntax/expr.rs:
```diff
@@ -1093,21 +1093,35 @@ fn parse_primary_expression(p: &mut Parser) -> ParsedSyntax {
+				// import("foo");
+				// import("foo", { assert: { type: 'json' } });
+
+				// test_err import_invalid_args
```
This test file doesn't exist. Have you run `cargo xtask codegen`?

Boshen replied:
Done

Boshen commented:
Squashed codegen and rebased, ready to merge.

MichaReiser commented on crates/rslint_parser/test_data/inline/err/import_invalid_args.rast:
```diff
@@ -0,0 +1,150 @@
+  ┌─ import_invalid_args.js:2:45
+  │
+2 │ import("foo", { assert: { type: 'json' } }, "bar")
+  │                                             ^^^^^ unexpected
```
Nit: A more specific error message for the case where there are no arguments or too many arguments would be nice

Boshen commented:
I rewrote this with a loop to get much better diagnostics.
I have also added a test case that is not in test262: `import(...["foo"])`
The error messages are copied from babel.

ematipico commented on crates/rslint_parser/src/syntax/expr.rs:
```diff
@@ -1100,21 +1100,57 @@ fn parse_primary_expression(p: &mut Parser) -> ParsedSyntax {
+				let completed_args_list = args_list.complete(p, JS_CALL_ARGUMENT_LIST);
+				if args_count == 0 || args_count > 2 {
+					let err = p
+						.err_builder("`import()` requires exactly one or two arguments. ")
```
I think we might be able to give a better diagnostic by highlighting the arguments that need to be deleted. 

It's not correct to highlight ALL the arguments in case - for example - we have 4 of them, but we should highlight only the last two.

Boshen commented:
I've pushed a commit to show the exact error range
